AutomatticTracks/src/main/java/com/automattic/android/tracks/crashlogging/internal/SentryCrashLogging.kt

```diff
@@ -45,9 +45,8 @@ internal class SentryCrashLogging constructor(
             options.apply {
                 dsn = dataProvider.sentryDSN
                 environment = dataProvider.buildType
-                release = when (val releaseName = dataProvider.releaseName) {
-                    is ReleaseName.SetByApplication -> releaseName.name
-                    ReleaseName.SetByTracksLibrary -> null
+                (dataProvider.releaseName as? ReleaseName.SetByApplication)?.let {
```

Comment:
Hmm, couldn't you just

```kotlin
release = (dataProvider.releaseName as? ReleaseName.SetByApplication)?.name
```

Reply:
It's the same overlook I did in #206 😅. The version you provided **will** assign `null` to `release` if `dataProvider.releaseName` is not `SetByApplication`. And, in result, **will** overwrite the value Sentry SDK automatically assigned (what we wanted).

Reply:
Aah gotcha, you don't really want the value to be `null` when the cast doesn't succeed 👍

Reply:
> when dataProvider.releaseName is SetByTracksLibrary

You mean **isn't** ? 😅

Reply:
🤔 `ReleaseName` can be either `SetByTracksLibrary` or `SetByApplication`.

When `dataProvider.releaseName` **is** `SetByTracksLibrary` we should do nothing with `options.release`. I think that's the correct description, matching the implementation.

Reply:
Ah, I thought you were mentioning the same type that's in the code. All clear now, thanks!